sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Catalog.scala

```diff
@@ -157,6 +163,15 @@ trait OverrideCatalog extends Catalog {
   override def unregisterAllTables(): Unit = {
     overrides.clear()
   }
+
+  override def showTables(databaseName: Option[String]): Seq[String] = {
+    val (dbName, _) = processDatabaseAndTableName(databaseName, "")
```

Comment:
this logic isn't super obvious. can you write some inline comment explaining what's happening here?